tests/jballerina-unit-test/src/test/resources/test-src/expressions/checkedexpr/checked_expr_error_return_type.bal

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+type E1 distinct error;
+type E2 distinct error;
+
+function foo(int state) returns int|E1|E2 {
+    if (state == 0) {
+        return error("Custom Error E1");
+    } else if (state == 1) {
+        return error E2("Custom Error E2");
+    } else {
+        return state;
+    }
+}
+
+function bar() returns error? {
+    return error("Dummy error");
+}
+
+function checkedExprErrorReturnType(int state) returns int|error {
+    return check foo(state);
+}
+
+function testCheckedExprErrorReturnType() {
+    test:assertTrue(checkedExprErrorReturnType(0) is E1);
+    test:assertTrue(checkedExprErrorReturnType(1) is E2);
+    test:assertTrue(checkedExprErrorReturnType(5) is int);
+    test:assertFalse(checkedExprErrorReturnType(0) is E2);
+    test:assertFalse(checkedExprErrorReturnType(1) is E1);
+}
+
+function errorReturnTypeWithMultipleCheckedExpr(int state) returns int|error {
+    if (foo(state) == state) {
+        return check foo(state - 2);
+    } else {
+        return check foo(state);
+    }
```

Comment:
If possible shall we make the tests simple so that it can be easily understood and seen what it tests while considering all the cases?